components/sql/src/main/scala/pl/touk/nussknacker/sql/db/schema/JdbcMetaDataProvider.scala

```diff
@@ -32,7 +32,7 @@ class JdbcMetaDataProvider(getConnection: () => Connection) extends DbMetaDataPr
     Using.resource(getConnection()) { connection =>
       Using.resource(connection.prepareStatement(query)) { statement =>
         TableMetaData(
-          TableDefinition(statement.getMetaData),
+          Option(statement.getMetaData).fold(TableDefinition(Nil))(TableDefinition(_)),
```

Comment:
Does Nu is still functional when getMetaData returns null? Shouldn't we communicate it as an Error to user and fail fast instead of pretending that everything is ok?

Reply:
Yes, Nu is still functional :) What is more, for update queries there is no metadata passed through, so it would not work without this change

Reply:
Ok, so why you don't return `Option[TableMetada]` and write a comment that for updates it is `None`? I see some hidden assumptions here :)

Reply:
I couldn't see them, sorry :) The idea with `Option` is fair enough and lead to better error handling inside db components. I've wrapped `TableMetada` in to `Option` and added cases to handle error when it is missing inside select query. For update queries it is a valid value.